pkg/standalone-utils/contracts/relayer/SiloWrapping.sol

```diff
@@ -0,0 +1,106 @@
+
+import "./IBaseRelayerLibrary.sol";
+
+/**
+ * @title SiloWrapping
+ * @notice Allows users to wrap and unwrap Silo shareTokens
+ * @dev All functions must be payable so they can be called from a multicall involving ETH
+ */
+abstract contract SiloWrapping is IBaseRelayerLibrary {
+    using Address for address payable;
+    using SafeERC20 for IERC20;
+
+    function wrapShareToken(
+        IShareToken wrappedToken,
+        address sender,
+        address recipient,
+        uint256 amount,
+        uint256 outputReference
+    ) external payable {
+        if (_isChainedReference(amount)) {
+            amount = _getChainedReferenceValue(amount);
+        }
+
+        // Initialize the token we will be wrapping (underlying asset of shareToken)
+        IERC20 underlyingToken = IERC20(wrappedToken.asset());
+        // Initialize the corresponding Silo (Liquidity Pool)
+        ISilo silo = wrappedToken.silo();
+
+        // The wrap caller is the implicit sender of tokens, so if the goal is for the tokens
+        // to be sourced from outside the relayer, we must first pull them here.
+        if (sender != address(this)) {
+            require(sender == msg.sender, "Incorrect sender");
+            _pullToken(sender, underlyingToken, amount);
+        }
+
+        underlyingToken.safeApprove(address(silo), amount);
+
+        // the collateralOnly param is set to false because we want to receive interest bearing shareTokens
+        (, uint256 result) = silo.depositFor(address(underlyingToken), recipient, amount, false);
+
+        if (_isChainedReference(outputReference)) {
+            _setChainedReferenceValue(outputReference, result);
+        }
+    }
+
+    function unwrapShareToken(
+        IShareToken wrappedToken,
+        address sender,
+        address recipient,
+        uint256 amount,
+        uint256 outputReference
+    ) external payable {
+        if (_isChainedReference(amount)) {
+            amount = _getChainedReferenceValue(amount);
+        }
+        // Initialize the token we will be withdrawing
+        IERC20 underlyingToken = IERC20(wrappedToken.asset());
+        // Initialize the corresponding Silo (Liquidity Pool)
+        ISilo silo = wrappedToken.silo();
+
+        // The wrap caller is the implicit sender of tokens, so if the goal is for the tokens
+        // to be sourced from outside the relayer, we must first them pull them here.
+        if (sender != address(this)) {
+            require(sender == msg.sender, "Incorrect sender");
+            _pullToken(sender, wrappedToken, amount);
+        }
+
+        // No approval is needed here, as the shareTokens are burned directly from the relayer's account.
+        // Setting the amount to uint256(-1) informs Silo that we'd like to redeem all the relayer's shares.
+        // Ignore the return value which cannot be trusted. It does not include any fees assessed.
+        silo.withdraw(address(underlyingToken), type(uint256).max, false);
```

Comment:
Comment says (-1); code is "max". Could reconcile by using a constant:
uint256 private constant MAX_UINT256 = type(uint256).max;